depinject/CHANGELOG.md

```diff
@@ -0,0 +1,28 @@
+<!--
+Guiding Principles:
+
+Changelogs are for humans, not machines.
+There should be an entry for every single version.
+The same types of changes should be grouped.
+Versions and sections should be linkable.
+The latest version comes first.
+The release date of each version is displayed.
+Mention whether you follow Semantic Versioning.
+
+Usage:
+
+Change log entries are to be added to the Unreleased section from newest to oldest.
+Each entry must include the Github issue reference in the following format:
+
+* [#<issue-number>] Changelog message.
+
+-->
+
+# Changelog
+
+## [Unreleased]
+
+## 1.0.0-alpha.x
```

Comment:
should we do 1.0? or what else is missing?

Reply:
We could do v1, or wait until we test it with generics

Reply:
But it's definitely v1 ready imho

Reply:
what is the generics approach?

Reply:
Idk, it was said depinject was and wasn't working with generics. We've actually no testcase to affirm or contradict that.

My point was more if it doesn't work, and we want to support it and supporting it requires breaking changes, then best to not tag v1 just yet. I'm saying that while I'm all for tagging v1 now but that is something to have in mind, we don't want to get v2 in a year or have to create another api for it. But again I've made no research or tests in that matter so I really don't know about the support.

Reply:
oh okay, i see. lets open an issue for that. If a test case is easy we can give it a try otherwise we can tag v1 and wait for v2 for generics

Reply:
Ok, I'll do alpha 5 for now, and we can do v1 in some time after when we found out that.

Reply:
So, I think we have an issue with depinject because of the cosmossdk.io/api imports. In my test it wouldn't work on a v0.47 chain.

Reply:
we should figure something out there. Core became like types, somewhere things would be put when we didnt know where to put it and since now we are trying to be more methodical with location of things we are running into issues

Reply:
In my opinion it was fine is core as it was module related. However I do agree that it doesn't fit anymore the requirements of core.